Feature set gateway

toros11 opened the pull request:
Allow setting the default gateway for interfaces

```
openvdc run centos/7/lxc '{
  "interfaces":[{
    "type": "veth",
    "ipv4addr": "192.168.1.199",
    "gateway": "192.168.1.1"
  }]
}'
```

Note: this PR also includes #215, which should be merged before

b0r6 committed:
Added option for which clone method to use

b0r6 committed:
Use different clone method depending on config

b0r6 committed:
Added config option

b0r6 committed:
added image location to template

toros11 committed:
Set datastore and base image as separate parameters

Cloning from vCenter templates doesn't require any datastore as
it can look up the correct template by id. I also separated the
parts related to the machine into a separate file and cleaned up
some redundant variables/types

toros11 committed:
Let the executor configuration file handle vCenter flag

toros11 committed:
Restructure configuration/parameter handling

toros11 committed:
Refactor for vCenter/ESXi

vCenter can handle namin/registring of vm's through the clone
functionallity and does not need to go through these steps.

toros11 committed:
Added missing setting parameter

toros11 committed:
Parse the inner template message

toros11 committed:
Temporary disable metadrive related code

toros11 committed:
Update vendors

toros11 committed:
Fix mistakes in vendor update

toros11 committed:
Added field: id to esxi nic interface

toros11 committed:
Remove redundant function

toros11 committed:
Change nic types to match esxi/vcenter terminology

toros11 committed:
Store the nic as metadata regardless of type

toros11 committed:
Setup netowrk devices

toros11 committed:
Re-enable metadata code

toros11 committed:
Merge branch 'master' into feature-esxi-vcenter

toros11 committed:
Readded accedentally removed field and fix syntax

toros11 committed:
Add/Update esxi template

Created sample template for use with vCenter crated OVF templates

toros11 committed:
Added field for gateway

toros11 committed:
Set gateway if defined

toros11 committed:
Fetch gateway

toros11 committed:
Added missing vendor

toros11 committed:
Fix another missing dependency

toros11 committed:
Update test

toros11 committed:
Update model

toros11 committed:
Use get functions to avoid nil pointer

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Rename id -> network id

toros11 committed:
Update test case to include network id/esxi adapter type

toros11 committed:
Skip if the machine already has a nic inside the network

toros11 committed:
Fix typo

toros11 committed:
Update model

toros11 committed:
Forgot to update schema

toros11 committed:
Added missing character

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Update model

toros11 committed:
Update vendor file

toros11 committed:
Wrote function to capture stdout

Vmwares library prints the output of api calls directly to stdout
and only returns an error code of wether the requests was able to
be sent succesfully or not. This lets us use the actual output of
the request to decide whether the request was succesfull or not

toros11 committed:
Parse device info to json format

toros11 committed:
Added missing dependency

toros11 committed:
Refactored the device check into a function

toros11 committed:
Wrote a function that checks if floppy device exists before adding

Currently openvdc uses fd0 to as mount point for metadata, here
we make sure that it gets attached correctly, and skips adding
a new device if one already exists

toros11 committed:
go fmt

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Set network/gateway/prefix for instance network

toros11 committed:
Fix order of parameters

toros11 committed:
Added some tests to check if gateway was succesfully set

toros11 committed:
Set esxi base image parameters directly in image structure

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Merge branch 'feature-set-gateway' of https://github.com/axsh/openvdc into feature-set-gateway

toros11 committed:
Fix some tests that i forgot to update

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Fix typo

toros11 committed:
Use the Timeout function and quote cmd to prevent word splitting

toros11 committed:
Fix some mistakes in template files

toros11 committed:
Create test that uses a base image without any devices attached

Also did some refactored the parameters into separate variables to
make it more readable

toros11 committed:
Allow image attributes to be overwritten

toros11 committed:
Added error type for govc api requests

Simply return when device.info fails.

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Added gateway parameter to Nic struct

toros11 committed:
Simulate the global ip used to test gateway

toros11 committed:
Change the simulated global ip to make it more clear that it's a different network

toros11 committed:
Move tests

toros11 committed:
Ping from global ip instnace

Removes dependency on openvdc console, if gateway is not set
the instance will not be able to reply to global ping requests.

toros11 committed:
Fix function call

toros11 committed:
Use correct ip calls and delete interface on destroy

toros11 committed:
Change to assignable mac address

toros11 committed:
Set global ip to a link-local type address to avoid possible default route problems

toros11 committed:
Remved exit call used for debugging

toros11 committed:
Reverse the link local ip, it wasn't the problem

toros11 committed:
Merge branch 'master' into feature-esxi-vcenter

toros11 committed:
Merge branch 'master' into feature-set-gateway

toros11 committed:
Merge branch 'master' into feature-esxi-vcenter

toros11 committed:
Merge branch 'master' into feature-set-gateway

toros11 committed:
Update sample config

unakatsuo reviewed (changes requested):
Please rename ``gateway`` -> ``ipv4gateway`` like that to clarify to put IPv4 address.  IPv6 gateway address parameter can be added later.

toros11 committed:
Merge branch 'master' into feature-esxi-vcenter

toros11 committed:
Merge branch 'master' into feature-set-gateway

toros11 committed:
Close/restore file descriptors before function return

toros11 committed:
fix typo

toros11 committed:
Don't use defer to close writer

The writer needs to close explicitly in order for io.Copy() to
return something, the writer needs to stay open until the buffer
is returned

toros11 committed:
Added unit test for captureStdout() function

toros11 committed:
Added modules subfield to config files

Using vCenter as the endpoint means that there is possibilities of
integration with a datacenter. This should make the management of
future features easier

toros11 committed:
Merge branch 'master' into feature-esxi-vcenter

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Revert and add prefix to vcenter flag

toros11 committed:
Merge branch 'feature-esxi-vcenter' into feature-set-gateway

toros11 committed:
Rename gateway -> ipv4gateway

Metallion commented:
Just like #215, merging since @b0r6 has approved, @unakatsuo's not available for review and there is a commit addressing his comment.
